warehouse/legacy/api/simple.py

```diff
@@ -47,6 +49,16 @@ def simple_index(request):
     return {"projects": projects}
 
 
+def get_sort_key_file(file):
+    return (
+        file.name,
+        # need to parse so 1, 10, 2 compare correctly
+        pkg_resources.parse_version(file.version),
+        # get same extensions together
+        file.filename[::-1]
+    )
```

Comment:
I don't think this needs to sort on `file.name`, as this is going to be the same for all  files within a project.

Instead of sorting on the reversed filename as a secondary sort, you can use `file.packagetype` instead.

At this point, the key function should be short enough that you can just do it inline, e.g.:

```python
    files = sorted(
        request.db.query(File)
        ...
        .all(),
        key=lambda f: (pkg_resources.parse_version(f.version), f.packagetype)
    )
```